plinth/utils.py

```diff
@@ -52,6 +52,11 @@ def non_admin_view(func):
     return func
 
 
-def is_user_admin(user):
+def is_user_admin(request, cached=False):
     """Return whether user is an administrator."""
-    return user.groups.filter(name='admin').exists()
+    if cached and request.session['cache_user_is_admin'] is not None:
```

Comment:
We have both made on mistake each :)

1) Checking for a value in the session will lead to a KeyError when there is no such key.  So, when I applied the patch and ran, it failed. Fix is to use `if 'key' in session`.
2) When anonymous users login, the cache is not getting cleared due to how Django sessions are reset during login.  To fix this, make the method return False if the user is not logged in by using `request.user.is_authenticated()` before any of the caching logic is applied.

You should certainly test more thoroughly :)